eiffel-syntax-and-usage/glossary.md

```diff
@@ -0,0 +1,102 @@
+
+ - [Terms Used in Documentation](#terms-used-in-documentation)
+   - [Pipeline](#pipeline)
+
+## Terms Used in Event Definitions
+These terms are either part of the event names in the Eiffel Vocabulary, or they are part of parameter names or values in the events. Users of Eiffel should be able to rely on these terms to be kept, unless a major change to event types are done.
+
+### Activity
+An _activity_ is any kind of action in a CI/CD system, normally triggered by an operation done in an SCM system or by a previous activity. The Eiffel protocol declares events related to activities to describe what is going on, what has finished, when did it happen and with what outcome.
+
+Activities are hierarchical. Activities could be whole [pipelines](#pipeline), or sequential or parallel parts thereof. Such pipeline parts could for example be build activities or test activities.
+
+An activity could also be an SCM operation such as a manual code review or automated source change check.
+
+Some activity types have their own specific events, such as EiffelTestCase\*Events notifying test case executions, while others are more generic and can be notified using EiffelActivity\*Events.
+
+#### Examples of events related to activities:
+- [EiffelActivityTriggeredEvent](../eiffel-vocabulary/EiffelActivityTriggeredEvent.md)
+- [EiffelActivityStartedEvent](../eiffel-vocabulary/EiffelActivityStartedEvent.md)
+- [EiffelActivityFinishedEvent](../eiffel-vocabulary/EiffelActivityFinishedEvent.md)
+- [EiffelTestCaseTriggeredEvent](../eiffel-vocabulary/EiffelTestCaseTriggeredEvent.md)
+- [EiffelTestCaseStartedEvent](../eiffel-vocabulary/EiffelTestCaseStartedEvent.md)
+- [EiffelTestCaseFinishedEvent](../eiffel-vocabulary/EiffelTestCaseFinishedEvent.md)
+
+### Artifact
+_Artifacts_ are items or software packages generated in a CI/CD [pipeline](#pipeline), for example a built binary or a Docker image. An artifact should be possible to identify using a purl (package URL). An artifact is often the subject of a test executed or a delivery performed within a CI/CD [pipeline](#pipeline).
+
+#### Examples of events related to artifacts:
+- [EiffelArtifactCreatedEvent](../eiffel-vocabulary/EiffelArtifactCreatedEvent.md)
+
+### Composition
+A _composition_ is a grouping of [artifacts](#artifact) and/or [source changes](#source-change). A composition is more or less the same as what is elsewhere sometimes referred to as a _baseline_.
+
+#### Examples of events related to compositions:
+- [EiffelCompositionDefinedEvent](../eiffel-vocabulary/EiffelCompositionDefinedEvent.md)
+
+### Confidence Level
+A _confidence level_ can be achieved for an [artifact](#artifact) or a [composition](#composition). It can be used to annotate that the artifact or composition has been tested up to a certain level, that it has reached a certain level of maturity, or that it has passed a certain criteria. Examples of confidence levels could be "smokeTestsOk", "releasable" or "released".
```

Comment:
It can also be applied to source changes, right? Since the link type supports SourceChange* events as subjects.

Reply:
Yes. Will add that.